### ISSUE TITLE
Client detects being offline

### ISSUE THREAD

**Issue by elsirion** [fedimint/fedimint#3496]
Currently client state machines just periodically re-try network requests when they notice that they cannot reach the federation. This can be wasteful if many are running in parallel and also doesn't give good feedback to the user.

The client should instead detect if it's connected to fewer than a threshold of guardian servers and make this fact available (internally and externally) in a way that can be subscribed to (likely a [`tokio::sync::watch`](https://docs.rs/tokio/latest/tokio/sync/watch/index.html) channel). 

This could then be used by the state machine executor to stop execution till we are online again, as by definition no progress can be made otherwise. *Important:* now that state machines don't trigger reconnects anymore another background task that tries to reconnect would be required.

**Comment by justinmoon** [comment]
This would be useful for apps integrating `fedimint-client`. We could have `Client::get_connection` and `Client::subscribe_connection` methods like we have for balances. If we could surface "peer id 1 and peer id 3 are offline" that would be useful information to have.

**Comment by justinmoon** [comment]
Would it make sense to also expose consensus problems (websocket RPC added [here](https://github.com/fedimint/fedimint/pull/2375)) in this or a similar manner?

**Comment by elsirion** [comment]
We already back-off API reconnects, so less pressing now, still nice to have in the future.
